tests/unittests/PartitionerTest.cpp

```diff
@@ -525,41 +525,124 @@ static void createSimpleModule(Module &mod) {
   (void)save;
 }
 
+/// Test using user-defined backends for heterogeneous partition.
 TEST_F(PartitionerTest, SimpleHeterogeneousPartitioning) {
-  {
-    createSimpleModule(mod_);
-    BackendWithoutSub backendWithoutSub1, backendWithoutSub2;
-    BackendWithoutMul backendWithoutMul1, backendWithoutMul2;
-    // Create two backends which support different ops, then do the partition by
-    // assigning the ops to the corresponding abackends.
-    std::vector<Backend *> backends;
-    backends.emplace_back(&backendWithoutMul1);
-    backends.emplace_back(&backendWithoutMul2);
-    backends.emplace_back(&backendWithoutSub1);
-    backends.emplace_back(&backendWithoutSub2);
-    std::vector<DeviceInfo> devices = {{3072, "Interpreter"},
-                                       {3072, "Interpreter"},
-                                       {3072, "CPU"},
-                                       {3072, "CPU"}};
-    auto partitioner =
-        Partitioner(&mod_, devices, backends, /* saturateHost */ true);
-    CompilationContext cctx;
-    auto err = partitioner.Partition(cctx);
-    EXPECT_FALSE(errToBool(std::move(err)));
-    DAGListTy myList = std::move(partitioner.getPartitionResult());
-    ASSERT_EQ(mod_.getFunctions().size(), 3);
-    ASSERT_EQ(myList.size(), 1);
-    ASSERT_TRUE(checkSaveNode(mod_));
-
-    for (auto &dag : myList) {
-      for (auto &node : dag.nodes) {
-        // Although the saturateHost is set true, no saturating the host in
-        // heterogeneous partiton.
-        ASSERT_EQ(node->logicalDevices.size(), 1);
+  createSimpleModule(mod_);
+  BackendWithoutSub backendWithoutSub1;
+  BackendWithoutMul backendWithoutMul1, backendWithoutMul2;
+  // Create two backends which support different ops, then do the partition by
+  // assigning the ops to the corresponding abackends.
+  std::vector<Backend *> backends;
+  backends.emplace_back(&backendWithoutMul1);
+  backends.emplace_back(&backendWithoutMul2);
+  backends.emplace_back(&backendWithoutSub1);
+  std::vector<DeviceInfo> devices = {
+      {3072, "Interpreter"}, {3072, "Interpreter"}, {3072, "CPU"}};
+  auto partitioner =
+      Partitioner(&mod_, devices, backends, /* saturateHost */ true);
+  CompilationContext cctx;
+  auto err = partitioner.Partition(cctx);
+  EXPECT_FALSE(errToBool(std::move(err)));
+  DAGListTy dagList = std::move(partitioner.getPartitionResult());
+  ASSERT_EQ(mod_.getFunctions().size(), 3);
+  ASSERT_EQ(dagList.size(), 1);
+  ASSERT_TRUE(checkSaveNode(mod_));
+
+  int numOfInterpreter = 0;
```

Comment:
The names sounds odd. Maybe `numOfInterpreterNodes` and `numOfCPUNode`?

Could you also check that it is exactly the kinds of nodes (or the nodes) you expected to be on CPU and Interpreter respectively?